src/tile_cmd.h

```diff
@@ -47,6 +47,15 @@ struct TileInfo {
 	int z;          ///< Height
 };
 
+/** Track extended description for the 'land area information' tool */
+struct TrackDesc {
+	StringID type;   ///< Type of track on the tile
+	uint16 speed;    ///< Speed limit (bridges and track)
+	const char *grf; ///< NewGRF used for the track
+
+	TrackDesc() : type((StringID) 0x0), speed(0), grf(nullptr) {}
```

Comment:
I mean an actual constructor, with parameters. You can roll the GetGRFName function into it as well.

Also, this should be STR_NULL

Reply:
```suggestion
	TrackDesc() : type((StringID) 0x0), speed(0), grf(nullptr) {}
	TrackDesc(StringID type, uint16 speed_limit, const char *grf_name) : type(type), speed(speed), grf(grf_name) {}
```

Reply:
Ouch!! understood, now it makes sense. STR_NULL cannot be resolved in that file, tried including some header files but nothing. Force pushing the changes. I made a proper constructor pulling GetGRFName in, looks simpler now.